specs/grafana_event_handling_in_monitoring_integration.adoc

```diff
@@ -0,0 +1,316 @@
+{
+  "evalMatches": [{
+    "value":6961614848,
+    "metric":"collectd.dhcp42-170_lab_eng_blr_redhat_com.memory.memory-free",
+    "tags":null
+   }],
+   "message":"alert",
+   "ruleId":6,
+   "ruleName":"Panel Title alert",
+   "ruleUrl":"http://localhost:3000/dashboard/db/new_graphite?fullscreen\u0026edit\u0026tab=alert\u0026panelId=3\u0026orgId=1",
+   "state":"alerting",
+   "title":"[Alerting] Panel Title alert"
+}
+
+Alert from API request: http://{ip}:3000/api/alerts/6
+{
+  Id: 6,
+  Version: 0,
+  OrgId: 1,
+  DashboardId: 7,
+  PanelId: 3,
+  Name: "Panel Title alert",
+  Message: "alert",
+  Severity: "",
+  State: "alerting",
+  Handler: 1,
+  Silenced: false,
+  ExecutionError: " ",
+  Frequency: 5,
+  EvalData:- {
+    evalMatches:- [{
+      metric: "collectd.dhcp42-170_lab_eng_blr_redhat_com.memory.memory-free",
+      tags: null,
+      value: 6961614848
+    }]
+  },
+  NewStateDate: "2017-07-18T23:07:16+05:30",
+  StateChanges: 291,
+  Created: "2017-09-18T14:36:37+05:30",
+  Updated: "2017-07-19T04:37:11+05:30",
+  Settings:- {
+    conditions:- [{
+      evaluator:- {
+        params:- [ 6956963696 ],
+        type: "gt"
+      },
+      operator:- {
+        type: "and"
+      },
+      query:- {
+        datasourceId: 3,
+        model:- {
+          refId: "A",
+          target: "collectd.dhcp42-170_lab_eng_blr_redhat_com.memory.memory-free"
+        },
+        params: - ["A","5m","now"]
+      },
+      reducer:- {
+        params:[],
+        type: "avg"
+      },
+      type: "query"
+    }],
+    executionErrorState: "alerting",
+    frequency: "5s",
+    handler: 1,
+    message: "alert",
+    name: "Panel Title alert",
+    noDataState: "alerting",
+    notifications:[]
+  }
+}
+    
+```
+
+After receiving alert then it will converted to current tendrl alert structure. 
+```
+Alert_id
+node_id
+time_stamp
+resource
+current_value
+tags
+alert_type
+severity
+significance
+ackedby
+acked
+ack_comment
+acked_at
+pid
+source
+```
+
+The event payload will also contains details regarding the scope of the alert which will be
+retrieved by node-agent to store the alerts under proper path in etcd.
+Proper formatted message will be provided by MessageFormatter(Alerthandlers in present
+alerting module). The formatting will be specific to the tendrl object the alerting event is
+for and these messages will be added to payload of event.
+Note: The notification will display the message set by MessageFormatter.
+
+Once all the required details(grafana specific data and tendrl specific metadata) regarding the event
+is collected from all respective data sources(grafana and tendrl objects) it is passed as a
+message object to the node-agent socket. The alerting event is supposed to be notified to the user
+which will be done by the notifier component.
+
+Notifier functionalities are explained in https://github.com/Tendrl/specifications/pull/199
+
+=== Alternatives:
+
+None
+
+
+=== Data model impact:
+
+
```

Comment:
You will have to use the tendrl.monitoring namespace and define flows/objects under it, check node-agent or current alerting repo